cfme/utils/targeted_refresh.py

```diff
@@ -0,0 +1,48 @@
+# -*- coding: utf-8 -*-
+"""
+class for targeted refresh tests
+"""
+
+import re
+import attr
+
+
+@attr.s
+class TargetedRefresh(object):
+
+    check_strings = attr.ib(default={})
+    entities = attr.ib(default=[])
+    triggered_refreshes = attr.ib(default=[])
```

Comment:
Two concerns here:

a) setting default value to mutable object and then modify it is never a good idea because strange behavior may occur, see https://stackoverflow.com/questions/44141104/when-and-why-should-i-use-attr-factory
b) why would you even want to let caller set those internal variables? Please consider marking them read-only like this:

```python
_triggered_refreshes = attr.ib(init=False, default=attr.Factory(list))
```

See http://www.attrs.org/en/stable/examples.html